### PR TITLE
feat(Banner): add new component

### LINKED ISSUE #1757: feat: new Banner component
A new component needs to be created: `Banner` it is like an Alert but more marketing oriented.

### PR DESCRIPTION
**Pull request by QuiiBz**
## Summary

Closes #1757 

## Type

- Feature
- Enhancement
- Documentation
- Migration
- Refactor

#### What is expected?

Add a new Banner component.

Waiting for the following to be merged:
- https://github.com/scaleway/scaleway-ui/pull/2403
- https://github.com/scaleway/scaleway-ui/pull/2406

## Relevant logs and/or screenshots

Screenshot 2023-04-20 at 11 15 32

Screenshot 2023-04-20 at 11 15 35

Screenshot 2023-04-20 at 11 15 39

### CODE REVIEW

File: packages/ui/src/components/Banner/index.tsx
Comment (philibea) [inline packages/ui/src/components/Banner/index.tsx:17]: we don't have gradient inside our theme ?

Comment (philibea) [reply packages/ui/src/components/Banner/index.tsx:17]: https://github.com/scaleway/scaleway-ui/pull/2406

Comment (QuiiBz) [reply packages/ui/src/components/Banner/index.tsx:17]: This PR was created before the one that adds the gradient, I'll wait for it to be merged to update and use the correct gradient

Comment (QuiiBz) [reply packages/ui/src/components/Banner/index.tsx]: Done @philibea

File: packages/ui/src/components/Banner/index.tsx
Comment (matthprost) [inline packages/ui/src/components/Banner/index.tsx:116]: Can you use a buttonV2 here?

**Review by matthprost — CHANGES_REQUESTED** [review]

**Comment by PetmePant** [comment]
Just some issues that I found:

- Buttons should be medium size and not large
- Close icon button it seems that it has no hover/active states.
- Wrong name in one type option: info instead of intro


*Still trying to figure out a solution for the promotional dark mode version

**Comment by QuiiBz** [comment]
Should be good now @PetmePant @matthprost

**Review by PetmePant — APPROVED** [review]
All looks okay, the only thing that might be good to change before final merge is to make the row direction default option.

File: packages/ui/src/components/Banner/__stories__/Image.tsx
Comment (matthprost) [inline packages/ui/src/components/Banner/__stories__/Image.tsx:27]: You exported a png in a svg that's not the best way of doing and it breaks the structure of storybook

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/Banner/index.tsx
Comment (matthprost) [inline packages/ui/src/components/Banner/index.tsx:63]: Could you integrate an internal state so when we click on close button the banner will close even if the developper didn't managed the state. You can check Alert to see how it is already done :)

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/Banner/index.tsx
Comment (QuiiBz) [reply packages/ui/src/components/Banner/index.tsx]: Done!

**Pull request merged**
